pkg/api/types.go

```diff
@@ -1733,6 +1733,11 @@ func (k *KubernetesConfig) GetOrderedKubeletConfigStringForPowershell() string {
 	return strings.TrimSuffix(buf.String(), ", ")
 }
 
+// HasContainerd returns whether or not we need the containerd runtime
+func (k *KubernetesConfig) HasContainerd() bool {
+	return k.ContainerRuntime == KataContainers || k.ContainerRuntime == Containerd
```

Comment:
this could be misleading in the future if we assume this function to mean `k.ContainerRuntime == Containerd`, now I understand your original naming `needsContainerd`...  Maybe we can make it explicit in the comment that Kata needs containerd as well so it shows in the function info when hovering from different files, wdyt? Also so we don't think later that this is an unintentional bug...

Reply:
other than that lgtm

Reply:
I think `needsContainerd` actually is the best option here, with an appropriate comment